### PR TITLE
Feature/edge browser implementation

### PR DESCRIPTION
**Pull request by Harris-H**
## Description
Add Microsoft Edge browser automation support to pydoll library.

## Changes
- Add Edge browser class with Edge-specific configurations
- Add test suite for Edge browser functionality
- Update browser initialization to expose Edge support

## Testing
All tests passing, including Edge-specific test cases.

   This PR addresses issue #41 by adding support for Microsoft Edge browser.

### CODE REVIEW

File: pydoll/browser/edge.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import platform
+
+from pydoll.browser.base import Browser
+from pydoll.browser.managers import BrowserOptionsManager
+from pydoll.browser.options import Options
+
+
+class Edge(Browser):
+    def __init__(
+        self, options: Options | None = None, connection_port: int = 9222
```
Comment (thalissonvs) [inline pydoll/browser/edge.py:11]: Set the default port to None and use the Optional type, so the Base class will generate a random port. I modified the Chrome class since it was causing conflicts in multiple instances

File: pydoll/browser/edge.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        
+        # Add debugging parameters
+        options.add_argument(f'--remote-debugging-port={connection_port}')
+        options.add_argument('--remote-allow-origins=*')
+
+        # Set default user data directory if not already set
+        if not any('--user-data-dir=' in arg for arg in options.arguments):
+            user_data_dir = os.path.join(os.path.expanduser('~'), '.edge_automation')
+            os.makedirs(user_data_dir, exist_ok=True)
+            options.add_argument(f'--user-data-dir={user_data_dir}')
```
Comment (thalissonvs) [inline pydoll/browser/edge.py:32]: When you call the `super()` method, these settings will be applied by default, so you don't need to duplicate them here

File: pydoll/browser/edge.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        if options is None:
+            options = Options()
+            
+        # Add Edge-specific startup arguments
+        options.add_argument('--no-first-run')  # Disable first run experience
+        options.add_argument('--no-default-browser-check')  # Disable default browser check
+        options.add_argument('--disable-crash-reporter')  # Disable crash reporting
+        options.add_argument('--disable-features=TranslateUI')  # Disable translation UI
+        options.add_argument('--disable-component-update')  # Disable component updates
+        options.add_argument('--disable-background-networking')  # Disable background networking
```
Comment (thalissonvs) [inline pydoll/browser/edge.py:22]: We have a class to manage the options called `BrowserOptionsManager`.

Here's my suggestion:

You can create an Enum with the available browsers, pass the browser name to the `Browser` class, and then forward it to the `BrowserOptionsManager`. There, you can create a method to configure the specific Edge options and call it if the specified browser is Edge.

Another idea: You can create a specific `Options` class that inherits from Options and set a property called `browser`. So, if the user wants to use Edge, they will have to use the specific Edge Options class.

In `BrowserManagerOptions`, you can check the browser property and then call the specific methods to configure each browser.

What do you think?

If you proceed with this idea, you'll have to create a specific class for `Chrome` too, inheriting from the base Options class and setting the `browser` property to "chrome".

**Comment by thalissonvs** [comment]
Hi @Harris-H,

First of all, thanks for your contribution! This feature is amazing.

I just left some comments to maintain the Pydoll code style and ensure the separation of concerns.

File: pydoll/browser/edge.py (hunk 3 above)
Comment (Harris-H) [reply pydoll/browser/edge.py]: I think your idea is good. I adopted method 2 and set the browser_type attribute in the option class, and created two subclasses based on it: ChromeOptions and EdgeOptions.

File: pydoll/browser/edge.py (hunk 2 above)
Comment (Harris-H) [reply pydoll/browser/edge.py]: OK, I put the edge-specific parameter configuration in manager.py and select different parameters through the BrowserType enumeration class.

**Comment by Harris-H** [comment]
> Hi @Harris-H,
> 
> First of all, thanks for your contribution! This feature is amazing.
> 
> I just left some comments to maintain the Pydoll code style and ensure the separation of concerns.

You're welcome. I have completed the main revisions. There may still be some areas that need to be improved. If you have any questions, we can discuss them again.

**Comment by thalissonvs** [comment]
Thank's! I'll review as soon as possible :)
Since it's a breaking change, I'll do some refactor first before merge.

**Comment by thalissonvs** [comment]
Hi @Harris-H , how are you? Once you fix the tests, please let me know

**Comment by Harris-H** [comment]
> Hi @Harris-H , how are you? Once you fix the tests, please let me know

Ok, I fixed the code and now the test passes

File: pydoll/browser/base.py [hunk 4]
```diff
@@ -570,20 +575,18 @@ async def _execute_command(self, command: str):
...
-        if '--user-data-dir' not in [
-            arg.split('=')[0] for arg in self.options.arguments
-        ]:
-            self.options.arguments.append(f'--user-data-dir={temp_dir.name}')
+        """Prepares the user data directory if necessary."""
+        if '--user-data-dir' not in [arg.split('=')[0] for arg in self.options.arguments]:
+            # Check if this is an Edge browser
+            if isinstance(self.options, EdgeOptions):
+                # For Edge browser, use a fixed user data directory
+                user_data_dir = os.path.join(os.path.expanduser('~'), '.edge_automation')
```
Comment (thalissonvs) [inline pydoll/browser/base.py:583]: just one question, why this directory is fixed?

**Comment by thalissonvs** [comment]
Hi @Harris-H , I just checked the code again, and the tests are still failing. I've created a CONTRIBUTING.md file. If you need any help, please contact me

File: pydoll/browser/base.py (hunk 4 above)
Comment (Harris-H) [reply pydoll/browser/base.py:583]: OK, I think it's better to use a temporary directory for everything, I've changed that.

**Comment by Harris-H** [comment]
> Hi @Harris-H , I just checked the code again, and the tests are still failing. I've created a CONTRIBUTING.md file. If you need any help, please contact me

image
Sorry, all tests should pass now.

**Comment by Harris-H** [comment]
> > Hi @Harris-H , I just checked the code again, and the tests are still failing. I've created a CONTRIBUTING.md file. If you need any help, please contact me
> 
> image Sorry, all tests should pass now. @thalissonvs

**Comment by thalissonvs** [comment]
Hi @Harris-H , the tests are passing, but the ruff ci is still breaking :/

**Comment by Harris-H** [comment]
> Hi @Harris-H , the tests are passing, but the ruff ci is still breaking :/

Hello @thalissonvs， the CI check of ruff has been repaired.

**Comment by thalissonvs** [comment]
Nice @Harris-H , thank you so much. I'll update the documentation later :)

**Review by thalissonvs — APPROVED** [review]
